sklearn/preprocessing/imputation.py

```diff
@@ -350,19 +380,28 @@ def _dense_transform(self, X, valid_stats, valid_idx):
 
         X[coordinates] = values
 
-        features_with_missing_values = np.where(np.any
-                                                (mask, axis=0))[0]
-        imputed_mask = mask[:, features_with_missing_values]
-        if self.axis == 0:
-            self.imputed_features_ = valid_idx[features_with_missing_values]
-        else:
-            self.imputed_features_ = features_with_missing_values
-
         if self.add_indicator_features:
-            X = np.hstack((X, imputed_mask))
+            X = np.hstack((X, imputer_mask_))
 
         return X
 
+    def _preprocess(self, X, statistics):
```

Comment:
Sorry I couldn't come up with a better name for this helper. Any suggestions would be helpful. Thanks.